### ISSUE TITLE
don't support both options fsdp and  gradient_checkpointing . why?

### ISSUE THREAD

**Issue by yanmj0601** [tianweiy/DMD2#52]
# assert not (train_args.fsdp and train_args.gradient_checkpointing), "currently, we don't support both options. open an issue for details."
why??

**Comment by tianweiy** [comment]
Using pure PyTorch's FSDP would offer better compatibility with gradient checkpointing and potentially much faster training, especially with newer PyTorch versions (unfortunately, we're currently stuck on PyTorch 2.0.1 due to Accelerate).

I plan to implement an Accelerate-free version in the future. However, I'm currently occupied with interviews, so I don't have a clear timeline for this yet :(

**Comment by yanmj0601** [comment]
ok, thanks
